skllm/models/gpt_zero_shot_clf.py

```diff
@@ -96,23 +88,16 @@ def _get_prompt(self, x) -> str:
     def _predict_single(self, x):
         completion = self._get_chat_completion(x)
         try:
-            label = str(
-                extract_json_key(completion.choices[0].message["content"], "label")
-            )
+            label = str(extract_json_key(completion.choices[0].message["content"], "label"))
+        except KeyError:
+            label = ""
         except Exception as e:
             label = ""
 
         if label not in self.classes_:
             label = random.choices(self.classes_, self.probabilities_)[0]
-        return label
 
-    def fit(
```

Comment:
While `.fit()` is inherrited from the parent, its signature is slightly different.